add paad_cptac_2021 study

egarcialara opened the pull request:
# What?
- added new study [Liwei Cao et al. "Proteogenomic characterization of pancreatic ductal adenocarcinoma" Cell 2021](https://www.cell.com/cell/fulltext/S0092-8674(21)00997-1)
https://github.com/cBioPortal/datahub/issues/1481

:information_source:  See [README](https://github.com/cBioPortal/datahub/blob/paad_cptac_2021/public/paad_cptac_2021/README.md) for transformation process

Cancer studies updated in this pull request:
- paad_cptac_2021

# checks
For all pull requests:
- [x] Passes validation

For a new study (in addition to above):
- [x] Does study name and study ID follow our convention? e.g. Tumor_Type (Institue, Journal Year); brca_mskcc_2015
- [x] is study meta data complete? e.g. pmid, group of PUBLIC
- [x] were all samples profiled with WES/WGS? If not, is gene panel file curated?
- [x] are oncotree codes of all samples curated; Cancer Type and Cancer Type Detailed needs to be added in addition to Oncotree Code
- [x] clinical sample and patient data with meta files
- [x] mutations data with meta files
- [x] MAF is based on hg19
- [ ] ~~MAF with 2 isoforms: uniprot and mskcc~~ _Not applicable_
- [x] CNA data with meta files
- [ ] ~~CNA segment data with meta files~~ _Not applicable_
- [x] Expression data including z-scores with meta files
- [x] Case-lists for all profiles.
- [ ] Manual checking (Niki or JJ): Triage or private Portal link here

Testing instance:
https://triage.cbioportal.mskcc.org/study/summary?id=paad_cptac_2021

Issues:
- CNA data missing, currently not included
The gistic2 gene level data provided looks faulty to us (every row/gene has exactly the same value) http://linkedomics.org/data_download/CPTAC-PDAC/SCNA_log2_gene_level.cct

SRodenburg reviewed (changes requested):
Hi @egarcialara 

Some fixes that are required before merge:

data_clinical_patient.txt
- please put the priority value for `MEDICAL_CONDITION` to 0, or perhaps even better, remove this attribute since the data is not really suitable for display. @yichaoS what do you think?

data_clinical_sample.txt
- there are quite a few composite cells, i.e. `7;30;25`. What do these numbers mean (replicates?) and can they be generalized?

data_mirna.txt
- is this the same file as data_microRNA_quantification.txt? if so delete either one.

data_mrna_*
- Just a generic question: should the `Entrez_Gene_Id` be empty, 0 or NA in case of a missing value?

data_mrna_seq_rsem_zscores_ref_diploid_samples.txt
- the zscores seem to be off. I think there was a mistake in the calculations, see my message in Slack.

data_N_glycoproteome_quantification.txt
- I think the `ENTITY_STABLE_ID` you used here is not the most suitable. Perhaps do it in a similar way as the methylation?

data_phosphoprotein_quantification.txt
- also here perhaps we should make the `ENTITY_STABLE_ID` according to the gene symbol. What do you think?

data_protein_quantification_zscores.txt
- I think some z-scores may noy be correct. Will double-check.

egarcialara commented on public/paad_cptac_2021/meta_study.txt:
```diff
@@ -0,0 +1,9 @@
+type_of_cancer: paad
+name: Pancreatic ductal adenocarcinoma. (CPTAC, Cell 2021)
+description: Proteogenomic characterization of pancreatic ductal adenocarcinoma. Generated by CPTAC.
+short_name: Proteogenome of human PDAC (CPTAC Cell2021)
```
I left `short_name` as it would otherwise fail the validator.
I also kept the PDAC abbreviation as the cancer abbreviation in the paper title, and paad for the cancer named used in cbioportal.

SRodenburg reviewed (approved):
I am only getting this error on validation:
```
ERROR: -: No case list found with stable_id 'paad_cptac_2021_all', consider adding 'add_global_case_list: true' to the meta_study.txt file
```

@yichaoS I think this is resolved when you use your internal loader?

SRodenburg reviewed (commented):
The CNA values are indeed strange and this can't be correct. I suppose the authors made a mistake in their CNA calling pipeline? Does the publication say anything about these data?

In the current form this CNA data is not very useful, and we might as well remove them.

yichaoS commented:
@SRodenburg the "cases_all" issue should be resolved now.

egarcialara commented:
> The CNA values are indeed strange and this can't be correct. [...] Does the publication say anything about these data?

@SRodenburg not really, besides that they summarized the CNV data into gene-level copy number changes.

egarcialara commented:
Number of features in downloaded table: 
- CNA: 19906
- RNA-seq: 28057
- protein: 11662
- cRNA: 3979
- miRNA: 2416
- methylation: 13180
- N-glycoprotein: 33000
- phosphoprotein: 51469
- mutations (maf): 6382

SRodenburg commented:
In that case I would be in favor of removing the CNA discrete data from this study

yichaoS commented:
> In that case I would be in favor of removing the CNA discrete data from this study

@SRodenburg I removed the CNA data and marked it as an issue in description.

yichaoS commented:
> Number of features in downloaded table:
> 
> * CNA: 19906
> * RNA-seq: 28057
> * protein: 11662
> * cRNA: 3979
> * miRNA: 2416
> * methylation: 13180
> * N-glycoprotein: 33000
> * phosphoprotein: 51469
> * mutations (maf): 6382

@egarcialara everything matches except mutations, a dozen rows short. Is it because of the same reason (missing symbol or ID?) Genome nexus could handle rows even without entrez ID and/or hugo symbol, as fill those missing info back up, as long as the coordinates present. Could you pleaes include all the 6382 rows from download? And then I would do the re-annotation on it to fill up missing symbols/IDs.

egarcialara commented:
Hi @yichaoS 
Great, I have updated the README.
I also updated the file `data_mutations.txt`, it has 6327 rows (6382 initial, - 57 rows after translating coordinates from hg38 to hg19, +2 rows that had two changes included in the `HGVSp_Short` column).

SRodenburg commented:
you can't mix genome builds in a single study so I suppose this is the maximum achievable

yichaoS commented:
> Hi @yichaoS Great, I have updated the README. I also updated the file `data_mutations.txt`, it has 6327 rows (6382 initial, - 57 rows after translating coordinates from hg38 to hg19, +2 rows that had two changes included in the `HGVSp_Short` column). Let me know if you prefer the full table with the cases `NCBIBuild==GRCh38` in it too.

@egarcialara @SRodenburg Right, this is because of lift over. Sorry and yes, this is fine.

yichaoS commented:
@egarcialara @SRodenburg I took a look at linkomics and saw their segment-level CNA data actually looks usable. Here is the link: http://linkedomics.org/data_download/CPTAC-PDAC/SCNA_log2_segment_level.cct. I then had a discussion with @jjgao about using this data. @egarcialara could you try to add this data using this format: https://docs.cbioportal.org/5.1-data-loading/data-loading/file-formats#segmented-data The `num of marks` column can be left as empty.

egarcialara commented:
Hi @yichaoS 
I just added the cna segment data, with its meta file, and updated the README. The file orginally had 28079 rows, but during the transformation of coordinates from hg38 to hg19, 3241 were lost. The `num.mask` is filled with `0`.
